index.js

```diff
@@ -1,8 +1,26 @@
 'use strict';
+
+function setCase(str, strCase) {
+	return str['to' + strCase + 'Case']();
+}
+
 module.exports = function (str, plural, count) {
 	if (typeof plural === 'number') {
 		count = plural;
-		plural = str + 's';
+
+		var plur = 's';
+		// get the tail of the string to check if the case should be upper or lower
+		var strCase = str.slice(1) === str.slice(1).toUpperCase() ? 'Upper' : 'Lower';
+		var end = str.slice(-2);
+
+		if (/s|x|z|ch|sh/i.test(end)) {
```

Comment:
This seems incorrect. You're testing against the last two characters, but with one character tests, that means that the `s` eg could be the next to last one and still match.

```js
if (/(s|x|z|ch|sh)$/i.test(str)) {
```